File: C/1399-count-largest-group/main.go
package main

import "fmt"

func countLargestGroup(n int) int {
	M := map[int]int{}

	for i := 1; i <= n; i++ {
		t := i
		c := 0
		for t > 0 {
			c += t % 10
			t /= 10
		}
		M[c]++
	}
	max := -1
	count := -1

	for _, v := range M {
		if v > max {
			max = v
			count = 1
		} else if v == max {
			count++
		}
	}
	return count
}

func main() {
	fmt.Println("a")
}

File: C/1399-count-largest-group/main.rs
use std::collections::HashMap;
impl Solution {
  pub fn count_largest_group(n: i32) -> i32 {
    let mut m: HashMap<i32, i32> = HashMap::new();
    (1..=n).for_each(|v| {
      let mut v = v;
      let mut sum: i32 = 0;
      while v > 0 {
        sum += v % 10;
        v /= 10;
      }
      m.entry(sum).and_modify(|x| *x += 1).or_insert(1);
    });

    let mut max: i32 = 0;
    let mut cnt: i32 = 0;
    for (_, v) in m {
      if v > max {
        max = v;
        cnt = 1;
      } else if v == max {
        cnt += 1;
      }
    }
    cnt
  }
}
